src/MBrace.Azure.Management/Management.fs

```diff
@@ -0,0 +1,810 @@
+    static member Standard_D2       : VMSize = "Standard_D2"
+    static member Standard_D2_v2    : VMSize = "Standard_D2_v2"
+    static member Standard_D3       : VMSize = "Standard_D3"
+    static member Standard_D3_v2    : VMSize = "Standard_D3_v2"
+    static member Standard_D4       : VMSize = "Standard_D4"
+    static member Standard_D4_v2    : VMSize = "Standard_D4_v2"
+    static member Standard_D5_v2    : VMSize = "Standard_D5_v2"
+
+/// Azure subscription record
+[<NoEquality; NoComparison; AutoSerializable(false)>]
+type Subscription = 
+    { 
+        /// Human-readable subscription name
+        Name : string
+        /// Subscription identifier
+        Id : string  
+        /// X509 management certificate
+        ManagementCertificate : string
+        /// Azure service management url
+        ServiceManagementUrl : string
+    }
+
+/// Parsed PublishSettings record
+[<NoEquality; NoComparison; AutoSerializable(false)>]
+type PublishSettings =
+    {
+        /// Set of Azure subscriptions defined in PubSettings
+        Subscriptions : Subscription []
+    }
+
+    /// Look up subscription by id or name
+    member ps.Item (subscriptionId : string) =
+        ps.Subscriptions |> Array.find (fun s -> s.Id = subscriptionId || s.Name.Contains subscriptionId)
+
+    /// Parse publish settings found in given xml string
+    static member Parse(xml : string) : PublishSettings = 
+        let parseSubscription (elem : XElement) = 
+            let name = elem.Attribute(XName.op_Implicit "Name").Value
+            let id = elem.Attribute(XName.op_Implicit "Id").Value
+            let mc = elem.Attribute(XName.op_Implicit "ManagementCertificate").Value
+            let smu = elem.Attribute(XName.op_Implicit "ServiceManagementUrl").Value
+            {   Name = name;
+                Id = id;
+                ManagementCertificate = mc
+                ServiceManagementUrl = smu }
+
+        let doc = XDocument.Parse xml
+        let pubData = doc.Element(XName.op_Implicit "PublishData")
+        let pubProfile = pubData.Element(XName.op_Implicit "PublishProfile")
+        let subscriptions = [| for s in pubProfile.Elements(XName.op_Implicit "Subscription") -> parseSubscription s |]
+        { Subscriptions = subscriptions }
+
+    /// Parse publish settings from given local file path
+    static member ParseFile(publishSettingsFile : string) : PublishSettings =
+        PublishSettings.Parse(File.ReadAllText publishSettingsFile)
+
+[<AutoOpen>]
+module private ManagementImpl = 
+
+    let urlForPackage mbraceNugetVersionTag vmSize = 
+        sprintf "https://github.com/mbraceproject/MBrace.Azure/releases/download/%s/MBrace.Azure.CloudService-%s.cspkg" mbraceNugetVersionTag vmSize
+
+    let resourcePrefix = "mbrace"
+    let generateResourceName() = resourcePrefix + Guid.NewGuid().ToString("N").[..7]
+    let defaultMBraceVersion = System.AssemblyVersionInformation.ReleaseTag
```

Comment:
Correct. My original plan was for the management lib to have a separate version number, but setting an arbitrary version string as default is simply too impractical. Since MBrace.Azure.Management does depend on MBrace.Azure, I do not see why they shouldn't have matching version numbers.